Arriba/Arriba/Model/Query/IQuery.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace Arriba.Model.Query
 {
-    public interface IQuery<T>
+    public interface IQuery
```

Comment:
I was very tempted to do this.... just kept waffling on it.  Potentially have a Type field so that it's possible to upcast to an IQuery<T>?  I'm not sure.